rpath issue on Mac: the JNI module cannot find `libvineyard_client.dylib`

<!--
Thanks for your contribution! please review https://github.com/v6d-io/v6d/blob/main/CONTRIBUTING.rst before opening an issue.
-->

Describe your problem
---------------------

The jni module fails to run if we don't patch the generated shared library with `install_name_tools` on Mac:

```
java.lang.ExceptionInInitializerError
	at io.v6d.core.client.IPCClientTest.connect(IPCClientTest.java:53)
Caused by: java.io.IOException: Couldn't load library library vineyard-core-1.0-SNAPSHOT
	at io.v6d.core.client.IPCClientTest.connect(IPCClientTest.java:53)
Caused by: java.lang.UnsatisfiedLinkError:
libvineyard/java/core/target/classes/libvineyard-core-1.0-SNAPSHOT.dylib: dlopen(libvineyard/java/core/target/classes/libvineyard-core-1.0-SNAPSHOT.dylib, 1): Library not loaded: @rpath/libvineyard_client.dylib
  Referenced from: libvineyard/java/core/target/classes/libvineyard-core-1.0-SNAPSHOT.dylib
  Reason: image not found
	at io.v6d.core.client.IPCClientTest.connect(IPCClientTest.java:53)
```

Additional context
------------------

Add any other context about the problem here.

--- a/java/core/pom.xml
+++ b/java/core/pom.xml
@@ -172,6 +172,8 @@
           </compilerEndOptions>
           <linkerStartOptions>
             <linkerStartOption>-L ${JAVA_HOME}/lib/</linkerStartOption>
+            <linkerStartOption>-L /usr/lib/</linkerStartOption>
+            <linkerStartOption>-L /usr/local/lib/</linkerStartOption>
           </linkerStartOptions>
           <linkerEndOptions>
             <linkerEndOption>-lvineyard_client</linkerEndOption>
@@ -186,4 +188,53 @@
       </plugin>
     </plugins>
   </build>
+  <profiles>
+    <profile>
+      <id>linux</id>
+      <activation>
+        <os>
+          <family>unix</family>
+          <name>Linux</name>
+        </os>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>native-maven-plugin</artifactId>
+            <version>1.0-alpha-11</version>
+            <configuration>
+              <linkerMiddleOptions>
+                <linkerMiddleOption>-L /usr/lib64/</linkerMiddleOption>
+                <linkerMiddleOption>-L /usr/local/lib64/</linkerMiddleOption>
+              </linkerMiddleOptions>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+    <profile>
+      <id>macosx</id>
+      <activation>
+        <os>
+          <family>mac</family>
+        </os>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>native-maven-plugin</artifactId>
+            <version>1.0-alpha-11</version>
+            <configuration>
+              <linkerMiddleOptions>
+                <linkerMiddleOption>-Wl,-rpath,/usr/lib</linkerMiddleOption>
+                <linkerMiddleOption>-Wl,-rpath,/usr/local/lib</linkerMiddleOption>
+              </linkerMiddleOptions>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+  </profiles>
 </project>

--- a/java/modules/basic/pom.xml
+++ b/java/modules/basic/pom.xml
@@ -56,7 +56,6 @@
     <dependency>
       <groupId>io.v6d.core</groupId>
       <artifactId>vineyard-core</artifactId>
-      <version>${project.parent.version}</version>
     </dependency>
 
     <dependency>

--- a/java/modules/pom.xml
+++ b/java/modules/pom.xml
@@ -22,18 +22,13 @@
     <module>basic</module>
   </modules>
 
-  <dependencies>
-    <dependency>
-      <groupId>io.v6d.core</groupId>
-      <artifactId>vineyard-core</artifactId>
-      <version>${project.parent.version}</version>
-    </dependency>
-
-    <dependency>
-      <groupId>junit</groupId>
-      <artifactId>junit</artifactId>
-      <version>4.13.1</version>
-      <scope>test</scope>
-    </dependency>
-  </dependencies>
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>io.v6d.core</groupId>
+        <artifactId>vineyard-core</artifactId>
+        <version>${project.parent.version}</version>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
 </project>

